src/main/java/seedu/duke/models/logic/CompletePreqs.java

```diff
@@ -0,0 +1,149 @@
+package seedu.duke.models.logic;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * CompletePreqs checks which mods are unlocked once you finish a specific module
+ */
+public class CompletePreqs {
+    //Full list of mods with modulesWithPreqs
+    private HashMap<String, List<String>> modulesWithPreqs;
+    private ArrayList<String> addToModulesCompleted;
+
+    // To track modules that are already unlocked
+    private Set<String> unlockedModulesSet;
+
+    public CompletePreqs(HashMap<String, List<String>> inputMods) {
+        modulesWithPreqs = new HashMap<>(inputMods);
+        unlockedModulesSet = new HashSet<>();
+        addToModulesCompleted = new ArrayList<>();
+    }
+
+    /**
+     * Create a list of completed mods, hardcoded from startup.
+     *
+     * @param list
+     */
+    public void initializeCompletedMods(ModuleList list) {
+        addToModulesCompleted.addAll(list.getMainModuleList());
+        for (String mod : addToModulesCompleted) {
+            processModuleForUnlockingWithoutPrint(mod);
+        }
+    }
+
+    /**
+     * Prints what mods have been unlocked after input
+     *
+     * @param moduleCompleted
+     */
+    public void getUnlockedMods(String moduleCompleted) {
+
+
+        // Check prerequisites of the moduleCompleted
+        if (modulesWithPreqs.containsKey(moduleCompleted)) {
+            List<String> unmetPrerequisites = new ArrayList<>();
+            for (String preq : modulesWithPreqs.get(moduleCompleted)) {
+                if (!addToModulesCompleted.contains(preq)) {
+                    unmetPrerequisites.add(preq);
+                }
+            }
+            //Stops if a completedMod shouldn't be able to be completed with proper preqs
+            if (!unmetPrerequisites.isEmpty()) {
+                System.out.println(moduleCompleted +
+                        " cannot be marked as completed because of uncompleted prerequisites: "
+                        + unmetPrerequisites);
```

Comment:
Consider using a standardized Ui object to execute your outputs. Avoid long strings, convert it to constants.